llvm/lib/Target/AMDGPU/SIInsertWaitcnts.cpp

```diff
@@ -1824,7 +1824,14 @@ bool SIInsertWaitcnts::generateWaitcntInstBefore(MachineInstr &MI,
   // Verify that the wait is actually needed.
   ScoreBrackets.simplifyWaitcnt(Wait);
 
-  if (ForceEmitZeroFlag)
+  // When forcing emit, we need to skip non-first terminators of a MBB because
+  // that would break the terminators of the MBB.
+  auto CheckIfMBBNonFirstTerminators = [](MachineInstr &MI) {
+    if (!MI.isTerminator())
+      return false;
+    return MI.getParent()->getFirstTerminator() != &MI;
+  };
+  if (ForceEmitZeroFlag && !CheckIfMBBNonFirstTerminators(MI))
```

Comment:
Go back to what it was before? There should be no reason to use getFirstTerminator